### PR TITLE
Update CarPlay tracking icon when tracking state changes

### LINKED ISSUE #2097: Overview button should change to recenter icon when panning out of course tracking in CarPlay
When panning the map in CarPlay to disable user tracking mode, the Overview button should change to the arrow icon to match its actual behavior or recentering the map and enabling user tracking mode.

The main browsing map template does show and hide the overview button after panning, but the navigation map template does not. All it does is disable user tracking.

https://github.com/mapbox/mapbox-navigation-ios/blob/4274aced2d02de0d5f416d416aaab3b228d1257b/MapboxNavigation/CarPlayManager.swift#L607-L609 https://github.com/mapbox/mapbox-navigation-ios/blob/4274aced2d02de0d5f416d416aaab3b228d1257b/MapboxNavigation/CarPlayManager.swift#L613-L615

/ref #2048
/cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #2097 

cc @1ec5

### CODE REVIEW

File: MapboxNavigation/CarPlayManager.swift [hunk 1]
```diff
@@ -142,20 +153,24 @@ public class CarPlayManager: NSObject {
     /**
      The bar button that shows the selected route overview on the map.
      */
-    @objc public lazy var overviewButton: CPMapButton = {
-        let overviewButton = CPMapButton { button in
+    @objc public lazy var trackingStateButton: CPMapButton = {
```
Comment (1ec5) [inline MapboxNavigation/CarPlayManager.swift]: Maybe call it `userTrackingButton` by analogy with MapKit’s MKUserTrackingButton.

/ref mapbox/mapbox-gl-native#1093

File: MapboxNavigation/CarPlayManager.swift [hunk 2]
```diff
@@ -55,8 +55,19 @@ public class CarPlayManager: NSObject {
...
     }
 
+    var trackingStateObservation: NSKeyValueObservation?
+    
     public fileprivate(set) var mainMapTemplate: CPMapTemplate?
-    public fileprivate(set) weak var currentNavigator: CarPlayNavigationViewController?
+    public fileprivate(set) weak var currentNavigator: CarPlayNavigationViewController? {
+        didSet {
+            if let controller = currentNavigator {
+                trackingStateObservation = controller.observe(\.tracksUserCourse) { [weak self] (controller, change) in
```
Comment (1ec5) [inline MapboxNavigation/CarPlayManager.swift]: Do we need to invalidate the NSKeyValueObservation at some point, at least on older iOS versions?

Comment (frederoni) [reply MapboxNavigation/CarPlayManager.swift]: Good catch. It would leak on iOS <= 10

File: MapboxNavigation/CarPlayManager.swift [hunk 3]
```diff
@@ -77,6 +92,10 @@ public class CarPlayManager: NSObject {
      */
     @objc public let directions: Directions
     
+    deinit {
```
Comment (1ec5) [inline MapboxNavigation/CarPlayManager.swift]: This is redundant to the `deinit` above, causing builds to fail:

```
Invalid redeclaration of 'deinit'
```

**Pull request merged**